backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java

```diff
@@ -0,0 +1,828 @@
+package de.wirvsvirus.trackyourbed;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import de.wirvsvirus.trackyourbed.dto.request.CreateNewBed;
+import de.wirvsvirus.trackyourbed.dto.request.UpdateBed;
+import de.wirvsvirus.trackyourbed.dto.request.mapper.CreateNewBedMapper;
+import de.wirvsvirus.trackyourbed.dto.response.BedDto;
+import de.wirvsvirus.trackyourbed.dto.response.BedStateDto;
+import de.wirvsvirus.trackyourbed.dto.response.BedTypeDto;
+import de.wirvsvirus.trackyourbed.dto.response.mapper.BedDtoMapper;
+import de.wirvsvirus.trackyourbed.dto.response.mapper.BedStateDtoMapper;
+import de.wirvsvirus.trackyourbed.dto.response.mapper.BedTypeDtoMapper;
+import de.wirvsvirus.trackyourbed.entity.Bed;
+import de.wirvsvirus.trackyourbed.entity.BedState;
+import de.wirvsvirus.trackyourbed.entity.BedType;
+import de.wirvsvirus.trackyourbed.entity.Ward;
+import de.wirvsvirus.trackyourbed.excpetion.dependency.InvalidBedStateException;
+import de.wirvsvirus.trackyourbed.excpetion.dependency.InvalidBedTypeException;
+import de.wirvsvirus.trackyourbed.excpetion.dependency.WardMissingException;
+import de.wirvsvirus.trackyourbed.excpetion.resource.NoSuchBedException;
+import de.wirvsvirus.trackyourbed.persistence.BedRepository;
+import de.wirvsvirus.trackyourbed.persistence.BedStateRepository;
+import de.wirvsvirus.trackyourbed.persistence.BedTypeRepository;
+import de.wirvsvirus.trackyourbed.persistence.WardRepository;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("Tests for BedStateService")
+class BedServiceTest {
+
+  @Nested
+  @DisplayName("Test calls to createNewBed")
+  class CreateNewBedTests {
+
+    @Test
+    @DisplayName("Should call dependencies and return expected result when called.")
+    void shouldCallDependenciesAndReturnExpectedResultWhenCalled() {
+      // GIVEN
+      final String state = "state";
```

Comment:
You are reusing a vast amount of the variables in a lot of tests, namely: `state`, `type`, `ward`,` request`, `bed`, `bedType`,.... Could you move some them to fixtures, so the duplicate code is avoided? 🙂

Reply:
Done